File: csharp/Svix/Models/MessageListOptions.cs
using System;
using System.Collections.Generic;

namespace Svix.Models
{
    public sealed class MessageListOptions : ListOptions
    {
        public List<string> EventTypes { get; set; }

        public string? Channel { get; set; }

        public DateTime? Before { get; set; }

        public DateTime? After { get; set; }

        public bool? WithContent { get; set; }

        public string? Tag { get; set; }
    }
}

File: php/src/Api/MessageListOptions.php
<?php

// this file is @generated
declare(strict_types=1);

namespace Svix\Api;

class MessageListOptions
{
    public function __construct(
        public readonly ?int $limit = null,
        public readonly ?string $iterator = null,
        public readonly ?string $channel = null,
        public readonly ?\DateTimeImmutable $before = null,
        public readonly ?\DateTimeImmutable $after = null,
        public readonly ?bool $withContent = null,
        public readonly ?string $tag = null,
        public readonly ?array $eventTypes = null,
    ) {
    }
}
